### ISSUE TITLE
Embeddables cannot be placed in MULTISET

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12237]
The following test:

```java
@Test
public void testEmbeddablesInMultiset() {
    insertData();

    Result<Record3<KeyRecord, Integer, Integer>> b =
        println(ctx.select(multiset(select(B.KEY, B.X, B.Y).from(B).orderBy(B.KEY)))).fetchSingle().value1();

    assertEquals(3, b.size());
    assertEquals(asList(k1, 3, 4), b.get(0).intoList());
    assertEquals(asList(k2, 5, 6), b.get(1).intoList());
    assertEquals(asList(k3, 7, 8), b.get(2).intoList());
}
```

Is trying to place an embeddable record inside of `MULTISET`. The table `B` is generated as:

```java
public class B extends TableImpl<BRecord> {
    public static final B B = new B();
    private final TableField<BRecord, Integer> KEY1 = createField(DSL.name("KEY1"), SQLDataType.INTEGER.nullable(false), this, "");
    private final TableField<BRecord, Integer> KEY2 = createField(DSL.name("KEY2"), SQLDataType.INTEGER.nullable(false), this, "");
    public final TableField<BRecord, Integer> X = createField(DSL.name("X"), SQLDataType.INTEGER, this, "");
    public final TableField<BRecord, Integer> Y = createField(DSL.name("Y"), SQLDataType.INTEGER, this, "");
    public final TableField<BRecord, KeyRecord> KEY = Internal.createEmbeddable(DSL.name("KEY"), KeyRecord.class, true, this, KEY1, KEY2);
    ...
```

When running the above query, we're getting (in H2):

```sql
select (
  select coalesce(
    json_arrayagg(json_array("v0", "v1", "v2" null on null)),
    json_array(null on null)
  )
  from (
    select
      "B"."KEY1", "B"."KEY2" "v0", -- Wrong aliasing here
      "B"."X" "v1",
      "B"."Y" "v2"
    from "B"
    order by "B"."KEY1", "B"."KEY2"
  ) "t"
)
```

The embeddable can't be aliased that way. This is actually a known issue from embeddable types: https://github.com/jOOQ/jOOQ/issues/11975

**Comment by lukaseder** [comment]
The aliasing of embeddables might still be an unsolved problem, but if an embeddable is placed within a `MULTISET` context, then it should be emulated like nested rows, not like embeddables.

**Comment by lukaseder** [comment]
The updated H2 SQL generation is now:

```sql
select (
  select coalesce(
    json_arrayagg(json_array("v0", "v1", "v2" null on null)),
    json_array(null on null)
  )
  from (
    select
      json_array("B"."KEY1", "B"."KEY2" null on null) "v0",
      "B"."X" "v1",
      "B"."Y" "v2"
    from "B"
    order by "B"."KEY1", "B"."KEY2"
  ) "t"
)
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.2 (https://github.com/jOOQ/jOOQ/issues/12239)
